plugin-locationlayer/src/main/java/com/mapbox/mapboxsdk/plugins/locationlayer/LocationLayerPlugin.java

```diff
@@ -431,11 +439,13 @@ private void disableLocationLayerPlugin() {
    * position.
    */
   @SuppressWarnings( {"MissingPermission"})
-  private void setLastLocation() {
+  private void setLastLocation(int locationLayerMode) {
```

Comment:
why does the end developer need to pass in the mode? isn't this part of the state managed by the plugin itself?

Reply:
This method isn't exposed to the end dev, it's purpose is to attempt at getting a location value immediately when a mode is set.

Reply:
That intent is not clear from the method name or the parameter, I would advice coming up with a better name for those.

Reply:
Thanks for clarifying, I'm still a bit confused why you aren't moving this [line](https://github.com/mapbox/mapbox-plugins-android/blob/ae4dc1b3c7f6fba46871ab88f3a5340065c19b95/plugin-locationlayer/src/main/java/com/mapbox/mapboxsdk/plugins/locationlayer/LocationLayerPlugin.java#L179) to the top of the method and just checking against `this.locationLayerMode` instead. Would this break any business logic?

Reply:
Good point, initially testing and not seeing any issues with this approach.